tensorwaves/physics/helicityformalism/amplitude.py

```diff
@@ -1,246 +1,402 @@
+            )
+
+        return self._parameters[name]
+
+    def _initialize_parameters(self, recipe: dict) -> None:
+        for par in recipe["Parameters"]:
+            self._parameters[par["Name"]] = tf.Variable(
+                par["Value"], name=par["Name"], dtype=tf.float64
+            )
+
+    def _get_normalization_data(self) -> Optional[dict]:
+        return self._phsp_data
+
+
+class _NormalizedIntensity:
+    def __init__(self, recipe: dict, builder, kinematics):
+        self._model_data = builder._create_intensity(recipe["Intensity"])
+        self._model_phsp = builder._create_intensity(recipe["Intensity"])
+        dataset = kinematics.convert(builder._get_normalization_data())
+        # its important to convert the dataset to tf tensors (memory footprint)
+        self._norm_dataset = {x: tf.constant(y) for x, y in dataset.items()}
+        self._norm_volume = atfi.const(kinematics.phase_space_volume)
+
+    @tf.function
+    def __call__(self, dataset):
+        normalization = tf.multiply(
+            self._norm_volume,
+            tf.reduce_mean(self._model_phsp(self._norm_dataset)),
         )
+        return tf.divide(self._model_data(dataset), normalization)
 
 
-class _IncoherentIntensity(tf.keras.Model):
-    def __init__(self, recipe: dict):
-        super(_IncoherentIntensity, self).__init__(name="IncoherentIntensity")
+class _StrengthIntensity:
+    def __init__(self, recipe: dict, builder):
+        if isinstance(recipe, dict):
+            self.strength = builder._get_parameter(recipe["Strength"])
+            self.intensity = builder._create_intensity(recipe["Intensity"])
+
+    def __call__(self, dataset):
+        return self.strength * self.intensity(dataset)
+
+
+class _IncoherentIntensity:
+    def __init__(self, recipe: dict, builder):
         if isinstance(recipe["Intensities"], list):
-            intensity_recipes = [v for v in recipe["Intensities"]]
+            intensity_recipes = list(recipe["Intensities"])
             self.intensities = [
-                _create_intensity(x) for x in intensity_recipes
+                builder._create_intensity(x) for x in intensity_recipes
             ]
         else:
             raise Exception(
                 "Incoherent Intensity requires a list of intensities!"
             )
 
-    def call(self, x):
-        return tf.add_n([y(x) for y in self.intensities])
+    def __call__(self, dataset):
+        return tf.math.accumulate_n([y(dataset) for y in self.intensities])
 
 
-class _CoherentIntensity(tf.keras.Model):
-    def __init__(self, recipe: dict):
-        super(_CoherentIntensity, self).__init__(name="CoherentIntensity")
+class _CoherentIntensity:
+    def __init__(self, recipe: dict, builder):
         if isinstance(recipe["Amplitudes"], list):
-            amp_recipes = [v for v in recipe["Amplitudes"]]
-            self.amps = [_create_amplitude(x) for x in amp_recipes]
+            amp_recipes = list(recipe["Amplitudes"])
```

Comment:
Is this conversion to `list` necessary?

Reply:
https://github.com/ComPWA/tensorwaves/pull/56/commits/5f7937d96cda5d51460ec1900d0d84e13d2ddf24